controlplane/kubeadm/controllers/controller.go

```diff
@@ -673,3 +684,36 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	return nil
 }
+
+// isKubernetesVersionSupported finds out if the Kubernetes version is supported.
+// Kubernetes versions >= v1.22.0 for management clusters are not supported. We identify
+// management clusters by the existence of the KubeadmControlPlane CRD.
+func (r *KubeadmControlPlaneReconciler) isKubernetesVersionSupported(ctx context.Context, cluster *clusterv1.Cluster, version string) (bool, error) {
```

Comment:
Should I move this func behind the `internal.WorkloadCluster` interface?

Regarding unit tests: I would (after I now where the func is placed correctly) move up the creation of the metadata client and then implement the usual table tests with a metadata fake client.

Reply:
As discussed. I'll keep it as is